### ISSUE TITLE
[bitnami/keycloak] Can't import themes because initContainer doesn't work with private registries

### ISSUE THREAD

**Issue by spoilerdo** [bitnami/charts#17376]
### Name and Version

bitnami/keycloak 15.1.4

### What architecture are you using?

amd64

### What steps will reproduce the bug?

This is my .gotmpl file where I use a docker image from a private registry

```
service:
    type: NodePort

  ingress:
    enabled: true
    ingressClassName: "alb"
    hostname: {{ .Values.host }}
    path: /*
    annotations:
      alb.ingress.kubernetes.io/scheme: internet-facing
      alb.ingress.kubernetes.io/target-type: instance
      alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}, {"HTTPS": 443}]'
      alb.ingress.kubernetes.io/ssl-redirect: '443'
      alb.ingress.kubernetes.io/certificate-arn: {{ requiredEnv "AWS_CRT_ARN" }}
      alb.ingress.kubernetes.io/group.name: "test-01-external"
      alb.ingress.kubernetes.io/backend-protocol: "HTTP"
      alb.ingress.kubernetes.io/wafv2-acl-arn: {{ requiredEnv "AWS_WAF_ARN" }}
      alb.ingress.kubernetes.io/manage-backend-security-group-rules: "true"
      alb.ingress.kubernetes.io/healthcheck-protocol: HTTP
      alb.ingress.kubernetes.io/healthcheck-path: /
      elbv2.k8s.aws/targetGroupBinding: shared

  extraVolumes:
    - name: theme
      emptyDir: {}

  extraVolumeMounts:
    - name: theme
      mountPath: /opt/jboss/keycloak/themes/xen

  initContainers:
    - name: theme-template
      image: "{{ requiredEnv "CI_REGISTRY" }}/{{ requiredEnv "IMAGE_NAME" }}:{{ requiredEnv "IMAGE_TAG" }}"
      imagePullSecrets: nexuscreds-keycloak
      imagePullPolicy: Always
      command:
        - sh
      args:
        - -c
        - |
          echo "Copying theme ..."
          cp -R /xen/* /theme
      volumeMounts:
        - name: theme
          mountPath: /theme
```

### What is the expected behavior?

That the initContainer can authenticate successfully with the private registry and pull the image from it.

### What do you see instead?

I get the following errors when I try to do this with a private Docker registry

![image](https://github.com/bitnami/charts/assets/21689486/7d3a193e-7417-45e3-81fa-09eb77e76fff)

So first an unauthorized and then a ImagePullBackOff error.

**Comment by javsalgar** [comment]
Hi,

Could you confirm that the rendered YAML file has the proper `imagePullSecrets` section?

**Comment by spoilerdo** [comment]
So I am relativly new to Kubernetes so please keep that in mind. I think you are refering to the following:
![image](https://github.com/bitnami/charts/assets/21689486/331cb88e-fccf-42c3-a760-5bbd0faa3fc9)

This is the secret and I also already tried naming the imagePullSecrets to nexuscreds-keycloak-xential-keycloak, but it has the same result. The secret contains a .dockerconfigjson using the following yaml config to make the secret:

```
apiVersion: v1
kind: Secret
metadata:
  name: {{ .Values.secret.name }}-{{ .Chart.Name }}
data:
  .dockerconfigjson: {{ .Values.secret.dockerConfigJson }}
type: kubernetes.io/dockerconfigjson
```

**Comment by spoilerdo** [comment]
![image](https://github.com/bitnami/charts/assets/21689486/8db1f42a-b0bc-451b-8cb9-8e847df6ce9a)

Also when I check on AWS in the raw view of the deployed keycloak pod. I do not see any imagePullSecrets section

**Comment by spoilerdo** [comment]
Do I perhaps need to use the global.imagePullSecrets instead of putting the secret value in initContainers.imagePullSecrets?

**Comment by spoilerdo** [comment]
Alright I fixed it myself by using a custom Docker image instead of the basic bitnami/keycloak one. I just extend that image with a COPY to copy my theme to the themes folder inside keycloak. Then I push the image to the private registry and use the following config:

```
  image:
    registry: {{ requiredEnv "CI_REGISTRY" }}
    repository: {{ requiredEnv "IMAGE_NAME" }}
    tag: {{ requiredEnv "IMAGE_TAG" }}
    pullPolicy: Always
    pullSecrets:
      - MY_SECRET
```

The secret I made with this:

```
apiVersion: v1
kind: Secret
metadata:
  name: {{ .Values.secret.name }}-{{ .Chart.Name }}
data:
  .dockerconfigjson: {{ .Values.secret.dockerConfigJson }}
type: kubernetes.io/dockerconfigjson
```

**Comment by ssichynskyi** [comment]
@carrodher @spoilerdo may I ask you to re-open this ticket? Although this particular issue was solved by other means, the original problem is still relevant. Custom image is an over-do if one only need to customize themes. So, could you be so kind to either:
1. Specify where Init Container registry secrets should be.
OR
2. Fix this issue if this is still relevant.

Thank you very much in advance!

**Comment by carrodher** [comment]
Thank you for reaching out and for your interest in our project. We'd be happy to assist you, but to do so effectively, we'll need some more details. Could you please create a separate issue filling the Issue template?

Providing this information will help us better understand your situation and assist you in a more targeted manner.

**Comment by ssichynskyi** [comment]
@carrodher, thank you for your answer
> Thank you for reaching out and for your interest in our project. We'd be happy to assist you, but to do so effectively, we'll need some more details. Could you please create a separate issue filling the Issue template?

That's kinda overdo as everything is already described in this ticket. In a nutshell:
1. I use Bitnami Keycloak chart to run the Keycloak.
2. I want to customize Keycloak themes (replace certain files)
3. The most convenient way to do that is to create an InitContainer that contains customized themes and copies it to Keycloak volume.
Should be easy if docker image repo of this InitContainer is public. But with private repo it is either not possible or unclear how to do. (`imagePullSecrets` for `InitContainers` is not described in a Helm chart. It's unclear if it shall be listed elsewhere, e.g. `global.imagePullSecrets`)
Basically, everything that @spoilerdo described originally and in comments.

**Comment by ssichynskyi** [comment]
For all those who like me ended up here. Chart works fine with private registries for `initContainers`. You just need to provide secrets for them inside `global.imagePullSecrets`
